runtime/lua/vim/lsp/client.lua

```diff
@@ -671,13 +671,32 @@ function Client:_request(method, params, handler, bufnr)
   bufnr = resolve_bufnr(bufnr)
   log.debug(self._log_prefix, 'client.request', self.id, method, params, handler, bufnr)
   local success, request_id = self.rpc.request(method, params, function(err, result)
+    -- Per LSP, don't show ContentModified error to the user.
+    if err and err.code ~= lsp.protocol.ErrorCodes.ContentModified then
+      vim.notify(
+        string.format('[LSP][%s] %s: %s', self.name, err.code, err.message),
+        vim.log.levels.ERROR
+      )
+      self:write_error(err.code, err.message)
+      api.nvim_command('redraw')
+    end
+
     local context = {
       method = method,
       client_id = self.id,
       bufnr = bufnr,
       params = params,
       version = version,
     }
+
+    if log.trace() then
+      log.trace('default_handler', method, {
+        err = err,
+        result = result,
+        ctx = vim.inspect(context),
+      })
```

Comment:
IIUC, it looks like this PR centralizes the debug/trace-level logging here, and drops the user-facing messages where there is no UX benefit to them. 💯 for that.

Reply:
Yeah, still working through some details. There are a few layers at play here. We absolutely do want to log any error that comes from a server.

I've **not** dropped the user facing messages, that's preserved in line 676, but I'm worried that calling `vim.notify` for all errors may be too much. There are some handlers that ignore errors, not sure if these are intentional.

Reply:
> There are some handlers that ignore errors, not sure if these are intentional

We can always merge and then "burn down" those cases as we discover them :)

Reply:
>  but I'm worried that calling vim.notify for all errors may be too much. There are some handlers that ignore errors, not sure if these are intentional.

There are some where we shouldn't show errors to users. 4a09c178a19097c295521892c889f1f196fff100 shows an example + context of why.

(I didn't have time yet to take a look over the diff of this PR - can do that later if desired)

Reply:
Thanks, that's helpful. I think that sort of context belongs as a comment directly in the code.

Reply:
Since this is a spec issue, we could special case this error directly in `client.request`, like we already do for `ContentModified`.